ext/nnstreamer/android_source/gstamcsrc_looper.cc

```diff
@@ -0,0 +1,185 @@
+/**
+ * GStreamer Android MediaCodec (AMC) Source Looper
+ * Copyright (C) 2019 Samsung Electronics Co., Ltd. All rights reserved.
+ * Copyright (C) 2019 Dongju Chae <dongju.chae@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	  gstamcsrc_looper.cc
+ * @date	  19 May 2019
+ * @brief	  A looper thread to perform event messages between amcsrc and media codec
+ * @see		  http://github.com/nnsuite/nnstreamer
+ * @author	Dongju Chae <dongju.chae@samsung.com>
+ * @bug		  No known bugs except for NYI items
+ */
+
+#include "gstamcsrc.h"
+#include "gstamcsrc_looper.h"
+
+#include <assert.h>
+#include <jni.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <limits.h>
+
+#include <android/log.h>
+#define TAG "AMCSRC-looper"
+#define LOG(...) __android_log_print(ANDROID_LOG_INFO, TAG, __VA_ARGS__)
+
+/**
+ * @brief Looper constructor; it creates a looper thread
+ */
+Looper::Looper() 
+{
+  pthread_attr_t attr;
+
+  head = NULL;
+  running = FALSE;
+  num_msg = 0;
+
+  pthread_mutex_init(&mutex, NULL);
+  pthread_cond_init(&cond, NULL);
+
+  pthread_attr_init (&attr);
+  pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED); 
+
+  pthread_create (&thread, &attr, entry, this);
+}
```

Comment:
plz check destroy function (pthread_attr_destroy (&attr)) after creation.

Reply:
You're right. It will be updated.